### ISSUE TITLE
[BUG] Grid.diff from center to outer is broken for chunked inputs

### ISSUE THREAD

**Issue by dcherian** [xgcm/xgcm#522]
**Minimal Complete Verifiable Example**:

```python
import xarray as xr
import xgcm

ds = xr.Dataset(
    data_vars={"foo": ("zc", np.arange(9.))},
    coords={"zi": np.arange(10.), "zc": np.arange(10.)[:-1] + 0.5},
)
grid = xgcm.Grid(
    ds, 
    coords={"Z": {"center": "zc", "outer": "zi"}},
    boundary="fill", 
    fill_value=np.nan
)
grid.diff(ds.foo, "Z")  # works!
grid.diff(ds.foo.chunk(), "Z")  # error!
```

**What happened**:

Error is raised here:
```
if map_overlap:
    744     # Disallow situations where shifting axis position would cause chunk size to change
--> 745     _check_if_length_would_change(sig)
    747     mapped_func = _map_func_over_core_dims(
    748         func,
    749         args,
   (...)
    753         out_dtypes,
    754     )
```

**What you expected to happen**:

This should just work; the core dimension isn't chunked and map_overlap isn't necessary.

**Anything else we need to know?**:

**xgcm version and environment**:

<details>

xgcm 0.8.0

</details>

**Comment by dcherian** [comment]
I also don't see why it needs to fail if the data is chunked along 'Z'. This is a meaningful operation (the numpy version works!) so the chunking should be irrelevant.

**Comment by rabernat** [comment]
This is the same core issues as #518

**Comment by TomNicholas** [comment]
> This should just work; the core dimension isn't chunked and map_overlap isn't necessary.

I agree that with a single chunk along the core dim this should work. The fact it doesn't is a bug I'm trying to fix now (the bug was introduced when trying to generalise to make `cumsum` work see #518).

> I also don't see why it needs to fail if the data is chunked along 'Z'. This is a meaningful operation (the numpy version works!) so the chunking should be irrelevant.

However IIRC there was a real reason for this restriction, see [this comment of mine](https://github.com/xgcm/xgcm/pull/380#issuecomment-975907377), copied below

> After getting really confused by the details of how to handle a chunked core dimension that could potentially change size, @rabernat and I decided to restrict this PR so that if a grid ufunc's signature includes the positions "inner" or "outer", then chunking along core dimensions is not allowed. This simplifies the code considerably because you can then be sure that chunk sizes never change. We might be able to relax this in future, but I would want to see a convincing use case to justify it first.

But I'm now having trouble remembering / understanding what the precise problem was... :sweat_smile:

**Comment by TomNicholas** [comment]
#558 should fix this for the case of a single chunk along core dims.

However I might keep this issue open in order to discuss the more difficult problem of handling arbitrary chunks along core dims, for grid positions that imply chunk sizes change.

**Comment by dcherian** [comment]
Can you update the error to suggest rechunking to a single chunk along the dimension if possible?

**Comment by TomNicholas** [comment]
> Can you update the error to suggest rechunking to a single chunk along the dimension if possible?

Yes, good idea, will do.

**Comment by jbusecke** [comment]
Agreed! This will be very useful.

**Comment by hdrake** [comment]
Fixed by #558
